### PR TITLE
use class getter method instead of directly using class property for getting file-extension string

### PR DESCRIPTION
**Pull request by bassauer-storms-media**
+ My IDE also removed some trailing whitespaces from the main BladeOne file but the only real change done in this commit is using the method `getFileExtension()` instead of getting the property directly with `$this->fileExtension`

### CODE REVIEW

**Comment by bassauer-storms-media** [comment]
Btw: just wanted to let you know: I love your BladeOne port of the blade-engine. Its an awesome work and I love that you keep it up to date. Just great!!

**Comment by jorgecc** [comment]
I can understand why we need to use setter and getters (mainly to standardize the code) but calling a getter inside the class. Why?.  It is an overhead that gives nothing.   You could say: but we could add logic to the getter... but it's not, we will not add logic to the getter because (unless it is used correctly) it's a bad practice.

https://dev.to/jorgecc/more-about-setter-and-getter-and-oop-4cpn

If we use setter and getters inside Java, then the runtime optimizes it and removes it, so it is transparent and it generates zero overhead.  But PHP doesn't have such feature.  

So, adding a setter and getter = ok, using it inside the class = overhead.

File: lib/BladeOne.php [hunk 1]
```diff
@@ -2137,12 +2137,12 @@ public function getTemplateFile($templateName = '')
         $c = count($arr);
         if ($c == 1) {
         	// its in the root of the template folder.
-            return $this->locateTemplate($templateName.$this->fileExtension);
+            return $this->locateTemplate($templateName.$this->getFileExtension());
```
Comment (bassauer-storms-media) [inline lib/BladeOne.php]: 1/2 relevant part of this commit

File: lib/BladeOne.php [hunk 2]
```diff
@@ -2137,12 +2137,12 @@ public function getTemplateFile($templateName = '')
...
         } else {
             $file = $arr[$c - 1];
             array_splice($arr, $c - 1, $c - 1); // delete the last element
             $path = implode('/', $arr);
-            return $this->locateTemplate($path.'/'.$file.$this->fileExtension);
+            return $this->locateTemplate($path.'/'.$file.$this->getFileExtension());
```
Comment (bassauer-storms-media) [inline lib/BladeOne.php]: 2/2 relevant part of this commit

**Comment by bassauer-storms-media** [comment]
@jorgecc 

Writing a template engine (blade) for a template engine (php) is overhead its self :x

**Reason for the request:**
I would need to hook into the getFileExtension method with a trait - thats why I would love to see BladeOne use that getter - but if you do not want to do it, thats okay, then I will be adding the getTemplateFile as a copy to my own trait in order to use the getter. In that case you can just close that pull request without merge.

While interpret/parse files in order to eval template code into native language code I don't think using a getter instead of direct var access impacts speed in any relation. I think the speed advantage of not using the getter but direct access the var will be a 0.00x micro second thingy.
The processor uses preg_replace multiple times (I know, it cannot be done another way and thats fine) which is known to be relatively slow (compared to other replacing functions in PHP). I think this may have much bigger speed impacts then using a method instead of direct var access which makes it less necessary to even talk about speed impact by using a getter.

(What I am planing to do is add the possibility of adding multiple file extensions so that blade can handle mixed up .php and .blade.php files in one and the same project (because we have dozens of old projets I want ot migrate to my web boilerplate which uses bladeone - but I don't want to change dozens of .php files to .blade.php and vice versa I dont want to rename all my blade components and views I already have from .blade.php to .php (which would also cause my IDE to lose syntax highlightning for blade files)))

**Pull request closed without merging**